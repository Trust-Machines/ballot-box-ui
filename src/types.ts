export type NETWORK = 'mainnet' | 'testnet';

export interface USER_AUTH {
    signature: string,
    publicKey: string
}
