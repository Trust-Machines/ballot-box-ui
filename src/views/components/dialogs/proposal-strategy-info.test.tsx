import {render} from '@testing-library/react';
import ProposalStrategyInfo from './proposal-strategy-info';
import {Proposal} from '../../../types';

const proposal: Proposal = {
    'id': 8,
    'userId': 1,
    'spaceId': 13,
    'title': 'Proposal with token',
    'body': null,
    'discussionLink': null,
    'startTime': 1663833600,
    'endTime': 1663952400,
    'startBlock': 76783,
    'startBlockTip': '0x28bbdab8a7283992629bd1d0cec4d795bfb1cb376a6bed5f7d83b8e51b944547',
    'choices': ['For', 'Against', 'Abstain'],
    'network': 'mainnet',
    'strategy': 'sip-010-get-balance',
    'strategyOptions': {
        'symbol': 'USDA',
        'address': 'SP2C2YFP12AJZB4MABJBAJ55XECVS7E4PMMZ89YZR.usda-token',
        'decimals': 6
    },
    'status': 'on',
    'voteCount': 0,
    'voteStats': [{'choice': 'For', 'power': 0}, {'choice': 'Against', 'power': 0}, {'choice': 'Abstain', 'power': 0}],
}

test('1 Default render', () => {
    const view = render(<ProposalStrategyInfo proposal={proposal}/>);
    expect(view.container).toMatchSnapshot()
});
