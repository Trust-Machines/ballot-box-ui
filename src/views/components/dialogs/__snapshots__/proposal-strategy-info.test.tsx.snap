// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Default render 1`] = `
<div>
  <h2
    class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
  >
    <button
      aria-label="close"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-lc2yqr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </h2>
  <div
    class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
  >
    <div
      class="MuiBox-root css-1dobh3l"
    >
      <div
        class="MuiBox-root css-jmbq56"
      >
        Symbol
      </div>
       
      <div
        class="MuiBox-root css-1i5zo8a"
      >
        USDA
      </div>
    </div>
    <div
      class="MuiBox-root css-1dobh3l"
    >
      <div
        class="MuiBox-root css-jmbq56"
      >
        Contract address
      </div>
       
      <div
        class="MuiBox-root css-1i5zo8a"
      >
        SP2C2YFP12AJZB4MABJBAJ55XECVS7E4PMMZ89YZR.usda-token
      </div>
    </div>
    <div
      class="MuiBox-root css-1dobh3l"
    >
      <div
        class="MuiBox-root css-jmbq56"
      >
        Decimals
      </div>
       
      <div
        class="MuiBox-root css-1i5zo8a"
      >
        6
      </div>
    </div>
  </div>
</div>
`;
